_rules/SC4-1-2-aria-attr-valid.md

```diff
@@ -0,0 +1,114 @@
+---
+name: ARIA attribute is valid
+
+description: |
+ This rule checks that each aria- attribute specified is defined in ARIA 1.1
+
+success_criterion:
+- 4.1.2
+
+test_aspects:
+- DOM Tree
+
+authors:
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+Any attribute that starts with `aria-`.
+
+### Expectation
+
+Each target attribute is defined in [WAI ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/).
```

Comment:
I would question whether or not this is a WCAG violation as well.

Reply:
I haven't been able to found anything in a spec supporting that this a violation of WCAG. However, I agree with @WilcoFiers that when someone uses an 'aria-' attribute is because they want to use some ARIA state or property and not because they just felt like creating a new attribute. But this is something that needs to be clearly expressed in the assumptions.

Reply:
I suggest changing the link to `[WAI-ARIA 1.1, Definitions of States and Properties](https://www.w3.org/TR/wai-aria-1.1/#state_prop_def)`

Not either sure that this is an actual requirement violation, but we do have several of these types, e.g. testing that only roles defined in WAI-ARIA are used etc. 

The reason we agreed that we could have these as ACT rules was that by using something ARIA-related, authors indicate that they have identified a need for AT accommodation, and they intend to meet it. If they then don't manage to do it correctly, there will likely be a need for accommodation, that is mistakenly is still unmet. 

So it's kind of like a spell-checking service for WAI-ARIA.

Reply:
Actually, thinking about it, was this rule intended for something else, like checking that only supported states and properties are used? 
Then maybe it should be something along these lines: 

Applicability: 
Any [WAI-ARIA 1.1 state or property](https://www.w3.org/TR/wai-aria-1.1/#state_prop_def) that is specified on an HTML or SVG element.

Expectation: 
The target state or property is a [supported state or property](https://www.w3.org/TR/wai-aria-1.1/#supportedState) for the [semantic role](#semantic-role) of the element it is specified on. 
(and then something about elements without a semantic role)

Reply:
@annethyme thanks for the explanation that indeed some of our rules are `spell-checking` service for WAI-ARIA.

And nope, this rule is not limited to just states and properties.